dll/shellext/cabview/util.h

```diff
@@ -0,0 +1,147 @@
+/*
+ * PROJECT:     ReactOS CabView Shell Extension
+ * LICENSE:     GPL-2.0-or-later (https://spdx.org/licenses/GPL-2.0-or-later)
+ * PURPOSE:     Utility header file
+ * COPYRIGHT:   Copyright 2024 Whindmar Saksit <whindsaks@proton.me>
+ */
+
+#pragma once
+
+template<class H> static int ErrorBox(H hWnd, int Error)
+{
+    SHELL_ErrorBox(hWnd, Error);
+    return Error;
+}
+
+template<class T> static inline bool IsPathSep(T c)
+{
+    return c == '\\' || c == '/';
+}
+
+inline bool IsEqual(const SHCOLUMNID &scid, REFGUID guid, UINT pid)
+{
+    return scid.pid == pid && IsEqualGUID(scid.fmtid, guid);
+}
+
+inline HRESULT InitVariantFromBuffer(const void *buffer, UINT cb, VARIANT *pv)
+{
+    SAFEARRAY *pa = SafeArrayCreateVector(VT_UI1, 0, cb);
+    if (pa)
+    {
+        CopyMemory(pa->pvData, buffer, cb);
+        V_VT(pv) = VT_UI1 | VT_ARRAY;
+        V_UNION(pv, parray) = pa;
```

Comment:
Was it not possible to use the `CComVariant` in this shell extension?
That would be my preference, but I won't see that as a blocker for merging it if you have good reasons not to do it.

Reply:
I'm not sure if CComVariant would help me a lot here. This array needs to be this exact type to be ABI compatible with [propsys::InitVariantFromBuffer](https://learn.microsoft.com/en-us/windows/win32/api/propvarutil/nf-propvarutil-initvariantfrombuffer) (which we don't have yet and certainly not for NT5). `SHGetDataFromIDList` in shell32 is supposed to be "on the other side" but currently just has a broken Wine implementation.